docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,121 @@
+      or complex equipments as robotic arms or other PLCs
+  * The **Process Image** is a block in the memory, that reflects inputs and outputs of all connected **Sensors** and **Actuators**
+    * it contains an _input area_ and an _output area_, where both areas are arrays of **data points**
+    * each **data point** carries a value of an input or output
+    * it is managed by the RTS during the cyclic execution of the **Control Application**, as steps below:
+      * Step 1: RTS reads all inputs from Sensor's and Actuator's, and stores all read input values to the **Process Image's** _input area_ 
+      * Step 2: RTS executes once the control application, that operates on the **Process Image's** _input_ and _output area_
+      * Step 3: RTS reads all output values from the **Process Image's** _output area_, and writes all read values to Sensor's and Actuator's
+      * then, next cycle starts with Step 1
+
+* **SoC**-based (**S**ystem **o**n **C**hip) Embedded System:
+  * TODO
+
+* **Microcontroller**-based Embedded System:
+  * TODO
+
+## thin-edge device concept
+
+**thin-edge** facilitates IoT functionality to the device it is running on, as well as to devices that are connected to that device.
+All devices are categorized as below:
+  * the **main-device** is the device thin-edge is running on
+  * all devices connected to the **main-device** are referred as **external child-devices**
+  * each device has a unique **device-id**
+
+## thin-edge data concepts
+
+**thin-edge** provides different devices in a standardized representation to the cloud. Therefore **thin-edge** provides different kinds of data concepts.
+
+  * **Measurements**:
+    * contain numeric data produced by sensors (like temperature readings) or calculated data based on information from the **control application**.
+    * a **measurements** consists of one or more numeric **data points** and optionally meta information as names or units
+  * **Variables**:
+    * contain numeric data provided and used by the device, and can be sent to the device (as e.g. "set points")
+    * a **variable** consists of one or more numeric **data points** and optionally meta information as names or units
+  * **Events**:
+    * contain other real-time information from the sensor network, such as the triggering of a door sensor
+    * an **event** is triggered by a **data point** (e.g. when it changes to a specific value as `0` or `1`), and contains an optional message text.
+  * **Alarms**:
+    * similar to Events, but the user or operator of the system has to take action to resolve the alarm
+    * an **alarm** is triggered by a **data point** (e.g. when it changes to a specific value as `0` or `1`), and contains an optional alarm text.
+
+## thin-edge plugin concept
+
+### Plugin
+
+**thin-edge** realizes high level cloud functionality (e.g. _configuration management_ or _log management_) with **plugins**.
```

Comment:
I don't think cloud functionality is the right term there. I'd have written `thin-edge realizes device management from the cloud using plugins`

Reply:
Fair point. I refer it now as "Device Management functionality".